tools/clang/test/CodeGenDXIL/hlsl/shader-flags/tiled_res.hlsl

```diff
@@ -0,0 +1,19 @@
+// RUN: %dxc -T ps_6_0 %s | FileCheck %s
```

Comment:
I find the separation and naming of these tests a little opaque.  They test that the flag is set for the use of the clamp argument specifically, which isn't in the naming.  Also, the added `_01` ... numbering isn't nearly as good as having the name of the method tested.  If 'Sample' is a given, these could be `tiled_res_clamp` and `tiled_res_clamp_[cmp]_[grad|bias]`.  But even that might be unnecessary if all these tests could be the same test, with different methods triggered by different define and run lines.

I also think new cases need to be tested: make sure no clamp or using constant `0.0f` for clamp does not set the flag.

Also, all these variants could be implemented in one test with multiple RUN lines rather than a bunch of separate test files with duplicated run lines.  I don't know what others prefer, but IMO if you have a test that only varies something minor that could be varied in a run line, it's a lot easier to tell that's the case, see the breadth of coverage, and maintain the tests if written in a more templated way using multiple run lines.